python/cudf/cudf/core/dataframe.py

```diff
@@ -6239,6 +6240,146 @@ def interleave_columns(self):
             {None: libcudf.reshape.interleave_columns([*self._columns])}
         )
 
+    @_cudf_nvtx_annotate
+    def eval(self, expr: "str", inplace: "bool" = False, **kwargs):
+        """Evaluate a string describing operations on DataFrame columns.
+
+        Operates on columns only, not specific rows or elements.  This allows
+        `eval` to run arbitrary code, which can make you vulnerable to code
+        injection if you pass user input to this function.
+
+        Parameters
+        ----------
+        expr : str
+            The expression string to evaluate.
+        inplace : bool, default False
+            If the expression contains an assignment, whether to perform the
+            operation inplace and mutate the existing DataFrame. Otherwise,
+            a new DataFrame is returned.
+        **kwargs
+            See the documentation for :func:`eval` for complete details
+            on the keyword arguments accepted by
+            :meth:`~pandas.DataFrame.query`.
```

Comment:
This seems copy-pasted out of place. Does this need to refer to `cudf.DataFrame.query`? Even so, does that behavior align with this? That method in cudf doesn't take any of the special keywords that pandas accepts.

Reply:
Yeah I'll need to fix this, thanks.